core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -51,8 +55,18 @@ func NewOutgoingConnectorHandler(gc connector.GatewayConnector, config ServiceCo
 }
 
 // HandleSingleNodeRequest sends a request to first available gateway node and blocks until response is received
-// TODO: handle retries and timeouts
-func (c *OutgoingConnectorHandler) HandleSingleNodeRequest(ctx context.Context, messageID string, payload []byte) (*api.Message, error) {
+// TODO: handle retries
+func (c *OutgoingConnectorHandler) HandleSingleNodeRequest(ctx context.Context, messageID string, req capabilities.Request) (*api.Message, error) {
+	// set default timeout if not provided for all outgoing requests
+	if req.TimeoutMs == 0 {
```

Comment:
@agparadiso I know I didn't ask you to do this originally, but could we also create a subcontext with this timeout plus some margin added to it?

Atm we don't get a response from the gateway if we time out which causes some small bugs; if we add this, the context would fire on line 101 and we'd at least get a response.

Reply:
added 👍🏼